src/project/internal/projectactionscontroller.cpp

```diff
@@ -501,11 +501,32 @@ bool ProjectActionsController::saveProjectToCloud(const CloudProjectInfo& info,
         return false;
     }
 
-    if (isCloudAvailable && project->isNewlyCreated()) {
-        Ret ret = openAudioGenerationSettings();
-        if (!ret) {
+    bool generateAudio = false;
+
+    bool isPublic = info.visibility == CloudProjectVisibility::Public;
+
+    if (saveMode == SaveMode::Save && isCloudAvailable) {
+        // Get up-to-date visibility information
+        auto scoreInfo = cloudProjectsService()->downloadScoreInfo(info.sourceUrl);
```

Comment:
please don't use auto here, and don't ignore possible errors, print them to the log